### PR TITLE
feat: add parse_geometry_obj function

### LINKED ISSUE #8: Parsing method for `Geometry`
Hi,

Right now I am in situation where I'd like to parse a PostGIS geometry into the Geometry models from this library. However, I don't know upfront which geometry type I'm dealing with.

I just wondered if it could be this library's responsibility of providing a method for parsing a geometry and returning the contents with the correct model. What do you think?

If so, I've been back and forth on where to do this. I think I've settled on putting a new function in `utils.py` or `parser.py` called `parse_geometry_obj` (same naming convention as pydantic `parse_obj`)  The function would return a `Geometry` type which is a union of all geometries.

Alternatively, we turn `_GeometryBase` into `Geometry` and override `parse_obj`, but I come from a functional background so inheritance like that feels odd and inflexible to me. In my world, we would get rid of the base class and add the geo interface to all the classes in the union.

Let me know what you think. I'd love to put up a PR. Should be really simple if we decide how to do it.

### PR DESCRIPTION
**Pull request by hawkaa**
Add a function parse_geometry_obj that will input an arbitrary GeoJSON geometry
object and output the correct Geometry model based on the 'type' field.

What do you think?

Fixes #8 .

### CODE REVIEW

File: geojson_pydantic/geometries.py [hunk 1]
```diff
@@ -52,6 +57,41 @@ class MultiPolygon(_GeometryBase):
     type: str = Field("MultiPolygon", const=True)
     coordinates: List[List[List[Coordinate]]]
 
+
 class GeometryCollection(BaseModel):
     type: str = Field("GeometryCollection", const=True)
-    geometries: List[Union[Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon]]
+    geometries: List[
+        Union[Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon]
```
Comment (hawkaa) [inline geojson_pydantic/geometries.py:64]: Should be `Geometry`.

File: geojson_pydantic/geometries.py [hunk 2]
```diff
@@ -52,6 +57,41 @@ class MultiPolygon(_GeometryBase):
...
+
+def parse_geometry_obj(obj) -> Geometry:
+    """
+    `obj` is an object that is supposed to represent a GeoJSON geometry. This method returns the
+    reads the `"type"` field and returns the correct pydantic Geometry model.
+    """
+    if "type" not in obj:
+        raise ValidationError(
+            [
+                ErrorWrapper(ValueError("Missing 'type' field in geometry"), "type"),
```
Comment (hawkaa) [inline geojson_pydantic/geometries.py:79]: I _think_ this is a decent way to report a ValidationError. Maybe good enough for a first try and possible to iterate on it if needed?

File: geojson_pydantic/features.py [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
 from .utils import BBox
-from .geometries import Point, MultiPoint, LineString, MultiLineString, \
-    Polygon, MultiPolygon
+from .geometries import (
+    Point,
+    MultiPoint,
+    LineString,
+    MultiLineString,
+    Polygon,
+    MultiPolygon,
```
Comment (drewbo) [inline geojson_pydantic/features.py:11]: I think these imports are unused now

Comment (hawkaa) [reply geojson_pydantic/features.py:11]: Looks like it. I can remove them. Adding mypy and/or pylint in github actions could help detect this in the future.

Comment (drewbo) [reply geojson_pydantic/features.py:11]: yup, just put up https://github.com/developmentseed/geojson-pydantic/pull/10

**Comment by hawkaa** [comment]
@drewbo fixed the unused imports :)

**Pull request merged**